Default dns server behavior changed

farnasirim opened the pull request:
Default dns server option was removed and reading dns addresses from HA instances etcd dir is now enforced.

farnasirim committed:
Default dns server behavior changed

Default dns server option was removed and reading dns addresses from HA instances etcd dir is now enforced.

remohammadi commented on dhcp/server.go:
```diff
@@ -95,6 +90,17 @@ func (h *DHCPHandler) fillPXE() []byte {
 func (h *DHCPHandler) ServeDHCP(p dhcp4.Packet, msgType dhcp4.MessageType, options dhcp4.Options) (d dhcp4.Packet) {
+	dns, err := h.datasource.DNSAddresses()
+	if err == nil {
+		h.dhcpOptions = dhcp4.Options{
```
If it's recalculated every time, can we change it to a local variable?

farnasirim replied:
No, I guess not. dhcp4.Options fields are not exposed and therefore not modifiable. The only way to give it a new option for dns seems to be re-instantiating it with the new option 6 []byte.

farnasirim committed:
Minor changes in dhcp server

Changed dhcpOptions to a local varaiable to prevent possible race conditions between two different calls.
Return nil packet when can't read dns addresses.

farnasirim committed:
Minor change. "go style"ing a part of dhcp/server.go
